google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+    String executorId = DefaultRunExecutor.getRunExecutorInstance().getId();
+    configurations.forEach(
+        configurationSettings ->
+            addEnvironmentVariablesToConfiguration(executorId, configurationSettings));
+  }
+
+  /**
+   * Adds the environment variables for the Google Cloud Libraries to the list of environment
+   * variables for {@code configuration} if they don't exist. If they exist, it replaces them.
+   */
+  private void addEnvironmentVariablesToConfiguration(
+      String executorId, RunnerAndConfigurationSettings configuration) {
+    ProgramRunner runner = ProgramRunnerUtil.getRunner(executorId, configuration);
+    if (runner == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    RunnerSpecificLocalConfigurationBit configurationSettings =
+        (RunnerSpecificLocalConfigurationBit) configuration.getConfigurationSettings(runner);
+    if (configurationSettings == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    Set<EnvironmentVariable> serviceAccountEnvironmentVariables =
+        commonPanel.getEnvironmentVariables();
+    List<EnvironmentVariable> configurationSettingsEnvVariables =
+        configurationSettings.getEnvVariables();
+
+    serviceAccountEnvironmentVariables.forEach(
+        serviceAccountEnvVar -> {
+          OptionalInt indexOpt =
+              IntStream.range(0, configurationSettingsEnvVariables.size())
+                  .filter(
+                      index ->
+                          serviceAccountEnvVar
+                              .getName()
+                              .equals(configurationSettingsEnvVariables.get(index).getName()))
+                  .findFirst();
+          if (indexOpt.isPresent()) {
+            configurationSettingsEnvVariables.set(indexOpt.getAsInt(), serviceAccountEnvVar);
+          } else {
+            configurationSettingsEnvVariables.add(serviceAccountEnvVar);
+          }
+        });
+    configurationSettings.setEnvironmentVariables(configurationSettingsEnvVariables);
+  }
+
+  @VisibleForTesting
+  public JTable getTable() {
+    return table;
+  }
+
+  /** Adds the Cloud Library environment variables to the selected App Engine run configurations. */
+  private class ApplyAction extends DialogWrapperAction {
+    private ApplyAction() {
+      super(CommonBundle.getApplyButtonText());
+    }
+
+    @Override
+    protected void doAction(ActionEvent event) {
+      addEnvironmentVariablesToConfiguration(getSelectedConfigurations());
+    }
+  }
+
+  /** The custom {@link JBTable} for the table of existing Google App Engine run configurations. */
+  private static final class ServerTable extends JBTable {
```

Comment:
Could we expand this name a bit to see what servers are in this table easier?

Reply:
Do you mean rename the JBTable to something like RunConfigurationTable?

Reply:
Exactly